eng/resolveContract.targets

```diff
@@ -1,17 +1,95 @@
+  </PropertyGroup>
+
   <PropertyGroup Condition="'$(IsSourceProject)' == 'true'">
-    <ContractProject Condition="'$(ContractProject)' == ''">$(LibrariesProjectRoot)$(MSBuildProjectName)/ref/$(MSBuildProjectName).csproj</ContractProject>
+    <ContractProject Condition="'$(ContractProject)' == ''">$(LibrariesProjectRoot)$(MSBuildProjectName)\ref\$(MSBuildProjectName).csproj</ContractProject>
     <HasMatchingContract Condition="'$(HasMatchingContract)' == '' and Exists('$(ContractProject)')">true</HasMatchingContract>
-    <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == ''">$(RefPath)/$(MSBuildProjectName).dll</ContractAssemblyPath>
+    <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == '' and
+                                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                                     $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">$(NetCoreAppCurrentRefPath)$(TargetFileName)</ContractAssemblyPath>
+    <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == ''">$([MSBuild]::NormalizePath('$(BaseOutputPath)', 'ref', '$(TargetFramework)-$(Configuration)', '$(TargetFileName)'))</ContractAssemblyPath>
 
     <!-- Disable API compat if the project doesn't have reference assembly -->
     <RunApiCompat Condition="'$(HasMatchingContract)' != 'true'">false</RunApiCompat>
   </PropertyGroup>
 
   <ItemGroup Condition="'$(HasMatchingContract)' == 'true'">
     <ResolvedMatchingContract Condition="Exists('$(ContractAssemblyPath)')" Include="$(ContractAssemblyPath)" />
+    <!-- If the contract doesn't exist in the default contract output path add a project reference to the contract project to resolve -->
+    <ProjectReference Condition="'@(ResolvedMatchingContract)' == ''" Include="$(ContractProject)">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+      <OutputItemType>ResolvedMatchingContract</OutputItemType>
+    </ProjectReference>
   </ItemGroup>
 
   <!-- intentionally empty since we no longer need a target -->
   <Target Name="ResolveMatchingContract" />
+  
+  <Target Name="SetILResourceReference"
+          BeforeTargets="DisassembleIlasmResourceFile"
+          Condition="'@(ResolvedMatchingContract)' != ''">
+    <ItemGroup>
+      <ILResourceReference Include="@(ResolvedMatchingContract)" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    If ContractTargetFramework is set, retrieve the ReferencePath from the contract
+    project and add it to ContractDependencyPaths.
+  -->
+  <Target Name="SetContractDependencyPathsFromReference"
+          BeforeTargets="RunMatchingRefApiCompat"
+          Condition="'$(HasMatchingContract)' == 'true' and '$(ContractTargetFramework)' != ''">
+    <MSBuild Projects="$(ContractProject)"
+             Properties="TargetFramework=$(ContractTargetFramework);Configuration=$(Configuration);BuildProjectReferences=false"
```

Comment:
Due to global properties, this will result in another evaluation of the referenced project.   If I follow the logic here we have two cases.
1.  ContractAssemblyPath exists, in this case it needs to be closure complete, so we can just use this directory for references.
2. ContractAssemblyPath doesn't exist.  We need to add a ProjectReference to ContractProject.  We should be able to return the ReferencePathDirectories on this item without another call to the project.

Reply:
Thanks, I removed the logic entirely as there is no need for such a target anymore.